src/helper/reactionRole.ts

```diff
@@ -0,0 +1,292 @@
+  } catch (error) {
+    serverLogger(
+      "internal-error",
+      "Unable to add Data to database and/or cache",
+      error.message
+    );
+  }
+};
+
+const refreshRoleData = async (roleData: roleSchema) => {
+  try {
+    const db = await (await getDbClient()).db().collection("reaction-roles");
+    await db.updateOne(
+      { messageID: roleData.messageID },
+      { $set: { reactions: roleData.reactions } }
+    );
+    const resultNodeCache = await setRole(roleData);
+    if (!resultNodeCache) throw { message: "nodecache error" };
+  } catch (error) {
+    serverLogger(
+      "internal-error",
+      "Unable to update data on database and/or cache",
+      error.message
+    );
+  }
+};
+
+export const addReaction = async (
+  reaction: MessageReaction,
+  user: User | PartialUser
+) => {
+  try {
+    if (!user.bot) {
+      const roleData = await getRole(reaction.message.id);
+      if (roleData) {
+        let userTag = roleData.reactions.find(
+          (userData) => userData.id === user.id
+        );
+        if (userTag) {
+          return reaction.users.remove(user.id);
+        }
+        let roles = roleData.options;
+        roles.forEach(async (data) => {
+          let emoji;
+          const emojiRegex = new RegExp(`(\:.*\:)`);
+          if (emojiRegex.test(data.emoji)) {
+            emoji = (data.emoji.match(emojiRegex) as Array<string>)[0];
+            emoji = emoji?.substring(1, emoji.length - 1);
+          } else emoji = data.emoji;
+          if (emoji === reaction.emoji.name) {
+            let role = reaction.message.guild?.roles.cache.find(
+              (role) => role.id === data.roleID
+            );
+            let userAdd = reaction.message.guild?.members.cache.get(user.id);
+            if (role && userAdd) {
+              if (userAdd.roles.cache.find((r) => r === role)) return;
+              await userAdd.roles.add(role);
+              roleData.reactions.push({
+                id: user.id,
+                tag: user.tag!,
+                roleID: data.roleID,
+              });
+              await refreshRoleData(roleData);
+              serverLogger("success", "role added", ``);
+              const channel = reaction.message.guild?.channels.cache.find(
+                (ch: any) => ch.id === process.env.LOGGER_CHANNEL_ID
+              ) as TextChannel;
+              if (channel)
+                channel.send(
+                  createBasicEmbed(
+                    INFO.REACTION_ROLE_ADD(role, userAdd),
+                    "LOG_2"
+                  )
+                );
+            }
+          }
+        });
+      }
+    }
+  } catch (error) {
+    serverLogger("internal-error", "unable to assign roles", error.message);
+  }
+};
+
+export const removeReaction = async (
+  reaction: MessageReaction,
+  user: User | PartialUser
+) => {
+  try {
+    if (!user.bot) {
+      const roleData = await getRole(reaction.message.id);
+      if (!roleData) return;
+      if (roleData) {
```

Comment:
If we are checking if (!roleData) return; then no need to check for if (roleData) {
```suggestion

```

Reply:
Yes, will do that. I'll also change if(!user.bot){ } to if(user.bot) return;

Reply:
Its fine for user.bot as you are not using else. But here in roleData its redundant